### PR TITLE
Change how git submodule status is retrieved

### PR DESCRIPTION
**Pull request by marko-bekhta**
I couldn't find a way to somehow pass on which rev to run the submodule command, and also didn't see some file from which we could read the hash, sooo that's what I ended up with 🙈

### CODE REVIEW

**Comment by yrodiere** [comment]
Well, tests pass.

@marko-bekhta can you check the caching stuff? If we're good, feel free to merge.

File: src/main/java/io/quarkus/search/app/util/GitCloneDirectory.java [hunk 1]
```diff
@@ -190,13 +190,43 @@ public ObjectId currentSourcesLatestHash() {
...
+                        e);
+            } finally {
+                checkout(currentBranch);
+            }
+        } catch (IOException e) {
+            Log.error("Wasn't able to correctly determine the current branch: " + e.getMessage(), e);
         } catch (GitAPIException e) {
-            throw new RuntimeException("Failed to list git submodule status for a repository: " + git.getRepository()
-                    .getRemoteNames() + ". " + e.getMessage(), e);
+            Log.error("Wasn't able to correctly switch back to '" + currentBranch + "'" + e.getMessage(), e);
```
Comment (yrodiere) [inline src/main/java/io/quarkus/search/app/util/GitCloneDirectory.java:217]: Shouldn't these be re-thrown? Or does it cause problems with tests?

Comment (marko-bekhta) [reply src/main/java/io/quarkus/search/app/util/GitCloneDirectory.java]: I thought about it, and my idea was that this happens, give the app another chance to try and work with the latest commit from the `main` repository. But then there shouldn't be any obvious reasons for git not to be able to checkout that branch 🤷🏻‍♂️ 😄, so I've changed it to re-throw.

**Comment by yrodiere** [comment]
Merged, thanks. Now let's see how it goes on staging :)

**Pull request merged**